### PR TITLE
Some janitoring

### PR DESCRIPTION
**Pull request by akva2**
Nothing major, just some header scrubbing and std::size_t usage.

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: opm/output/eclipse/RestartIO.hpp [hunk 1]
```diff
@@ -24,11 +24,8 @@
 #ifndef RESTART_IO_HPP
 #define RESTART_IO_HPP
 
-#include <opm/output/eclipse/AggregateAquiferData.hpp>
```
Comment (bska) [inline opm/output/eclipse/RestartIO.hpp:27]: Don't you need a full definition of `Helpers::AggregateAquiferData` in scope?  The `save()` function takes an `optional<>` of that type as [an argument](https://github.com/OPM/opm-common/blob/c42f29c45f16906515ad6dfb1400ebc4416c869d/opm/output/eclipse/RestartIO.hpp#L93).

**Review by bska — COMMENTED** [review]
I don't think we're supposed to create `std::optional<IncompleteType>`, so by removing `AggregateAquiferData.hpp` from `RestartIO.hpp` I **believe** you're making the header dependent on inclusion order elsewhere.

File: opm/output/eclipse/RestartIO.hpp (hunk 1 above)
Comment (akva2) [reply opm/output/eclipse/RestartIO.hpp:27]: as it's a reference to the optional to it does not need to instance the template it seems.

```
#include "../opm/opm-common/opm/output/eclipse/RestartIO.hpp"

int main(int argc, char** argv)
{
  return 0;
}
```

builds fine.

Comment (bska) [reply opm/output/eclipse/RestartIO.hpp:27]: > as it's a reference to the optional to it does not need to instance the template it seems.

Hm, I see.  For what it's worth, [Compiler Explorer (godbolt.org)](https://godbolt.org/#z:OYLghAFBqd5TKALEBjA9gEwKYFFMCWALugE4A0BIEAZgQDbYB2AhgLbYgDkAjF%2BTXRMiAZVQtGIHgBYBQogFUAztgAKAD24AGfgCsp5eiyahUAUgBMAIUtXyKxqiIEh1ZpgDC6egFc2TA3cAGQImbAA5PwAjbFIpWQAHdCViFyYvX38DJJTnIRCwyLYYuJl7bEc8phEiFlIiDL8AnnLKtJq6ogKI6Nj4%2B1r6xqyWpUGu0J7ivpkASnt0H1JUTi5LAGZQ1F8cAGozda8EqokD3DMtAEELy9YOJQSWFd2ADX2Adhsr3Z/dsdIfE5diIDl9ru8ACI3G53bAPJ7YD5g367KLoby7GgQDBMMZ/IiYEAgdDHNKnQ4vIkg9bnCwANl26FmoOhkOhV1CRF2bBYoQgzKuZk%2BNxRLB8JAAVLtjqR9usIft6TQWd9fqRsEQlkxpURZWAwAcFUwfPR6DKVeCIVx5vRuABWfgBLg6cjobgeWy2P6LZaIjZ8chEbTW%2BYAaxA6zpADppO8LBYAJw8LR26QADgsPHWdsM3Gk/DYIDtWnITpdbq4/CUIBLQed1vIcFgKAwbASDFilGorfbjDiwCUrASSiQ6CIAgYRFi1YgUWD5CioTqAE9uAHF6xSMuAPJRXTYJxr/itjjCbdMeir%2BvkHBRHzADwSejV3j8HA8kySa%2BEdVOAgANzhedsHUA9xVWANOQqed6AIKJSBXLwcHnXUCELV95hoIwBwANQIbAAHdtwSZgjzkYQxAkTgykEYRlDUTRr30CxDGMUxPWsQw4OrSB5hJKoXwAWm3dZdkEgB1J8xPEkDdRYMSSMwYxnFQMSxkwQ1zGsWwLHLQDSFIAgcB4/lWgPKo3CYTxvCaQIrO6IoSmyZJUiEYZmnIHJXKYBzelKMy/yEDohhskYAqqYKJkKPyDDGTp3Ni8ZfOmUp5iUH0VikG17UdecK12dQ0zpQS6WkXZgFQVSIF1HwmFDWZdggfBiDIRV1h4WZ%2BDrHRZnmJBsBYHA4lM20uHzchC2LUs8u4Ksa0DYNevIcN2qjO06TTBMtDpHg6XWdZ3mkO0LBzUb1ly68Ky6xbG0QJtkDQdA2w7CgqGxJ7ez6bY2IAfRquqJ3oKdSBnOdrw3FcyIhrdd33Q9X3IE9mCIc9L3nW970fU0XwDd82K/F0f3MgCgOvECwKnMioNGl1YPgxCsAgrrDPQvhMOwpQ8MI4jSIR2jRHESQaPkeiNHnfQWiMEw0A4uw6ZMvjSSEISROkqSJNkhCFOwJThAIVTBPUzTZd0tSCTpw06a0mxrH4dB9MM4z4DSipidcCB3ASlpgkmRyZk8lyqi9gPcjSZKnNGV3AuqcZg4cN2Y86cP/bikLMg81OoqmCO0oy6jsq4B1psu7gCqKkqyu%2BkxGv%2B%2BrGuakhZX9TqFvrJaVrTKMiukLR3jtXaeHeBMk1zMaCyLEsyzt2b7Hm7qG3uiAWw%2Bl6u3e56%2BxAOo2H/NNBKr4BBNrwHgdB%2BdoavdclxhvdzLIpGzwvK9CewO8HyfHG32wD9gAJt8CF/M4QCL4XTk1QOBKmwhoLXjpghLcSEmaBhZkedmLBcL4SIiRJ0AZ%2BaUSFrIfmotGIun0KdNiMttKcXls7V0StcTcGEqJCS6sZLqDktrXWKkzYaXlNbHSWhuEW3lPLAkfDbaugdkZICvFwppEstZdOdlMDJ38l5IOoUPJqLDr7GKkc2hBVjho2KUcIpJR0SlRK8UjGjDMdFCxHUFhLEyg40aRcp6ulLtvXegl5IHxrgCOqDUmqEEbm1Dq1024F3GpNSeM1Kyz1rItMMEYtBrQsNIbMg96R0hyTId4o9zrF3LDPee8xF53WXhvTsb0ewvRAPrVAulMzrBLHQIG05qBgxdBfKG18dy33hgGB%2BKMn7o1fpjD%2BZE8aflWITABxNgHAVAuAym15qYwTgnA5cCCULIIwgIDmXNMG8xwfIPB1ECEixUGLJiEZWLSzEXLbiND%2BJpBVkwySpppKa3koJRSyl9bcONpQqwuk7aSKdjI%2BO0d5HBx9nYiOIdvLBy0fkcxiLoWmKsYovRCdIoqMsWnWyNik7opmLnJx%2BdR5uLiWXYqpVdgNN2E0ngUZ1hRgEcElqTcLDtRbqUvqA0hrUGSesCwMYeCZneHSLQJ0tAyC0AmUe0SJ5FOnvE6siS263XgEvR6VTXrdhXpvf8qAEgJB%2Bv%2BHgCYfpGCnGMH66hSonw6bOc%2BfTembn6XDOsQynqnhGWjb84z37Yymd/fGsz/6AJJiA/gYCIEI3WTAzZDNkLXlQqzBsWE0GcwwTzbB/BcGCwueRRQ1ziF6BALIKW7EQVcSiArWhAluAAHptwWF2K29SFCbagr0rER20jTKYrkR7KycL7JktUYHNIKLZ1ooRSnEx7RDE4tkQY0lS7/KZzjrY7O5LHG%2BiytSi6xSuAFQZWwJQprdhWoTFGO1cIuRctCc3CJPVBWDT6CNPM48pruKugk1un7loRneFGLadJDqZkVQmO0CZ9oFLPeqj9IZR4WBQx4%2BJAryD6RSK4aQQA%3D) agrees with you.  That said, I think I'd still prefer to include the header just to sidestep these discussions in the future.

Comment (akva2) [reply opm/output/eclipse/RestartIO.hpp]: restored.

**Comment by akva2** [comment]
jenkins build this please

**Pull request merged**